src/EditorFeatures/CSharpTest/MakeFieldReadonly/MakeFieldReadonlyTests.cs

```diff
@@ -0,0 +1,657 @@
+        public async Task FieldAssignedInMethodWithCompoundOperator()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private int [|_foo|] = 0;
+        int Foo(int value)
+        {
+            _foo += value;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldUsedWithPostfixIncrement()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private int [|_foo|] = 0;
+        int Foo(int value)
+        {
+            _foo++;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldUsedWithPrefixDecrement()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private int [|_foo|] = 0;
+        int Foo(int value)
+        {
+            --_foo;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task AssignedInPartialClass()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    partial class MyClass
+    {
+        private int [|_foo|];
+    }
+    partial class MyClass
+    {
+        void SetFoo()
+        {
+            _foo = 0;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task PassedAsParameter()
+        {
+            await TestInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private int [|_foo|];
+        void Foo()
+        {
+            Bar(_foo);
+        }
+        void Bar(int foo)
+        {
+        }
+    }
+}",
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private readonly int _foo;
+        void Foo()
+        {
+            Bar(_foo);
+        }
+        void Bar(int foo)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task PassedAsOutParameter()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private int [|_foo|];
+        void Foo()
+        {
+            int.TryParse(""123"", out _foo);
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task PassedAsRefParameter()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private int [|_foo|];
+        void Foo()
+        {
+            Bar(ref _foo);
```

Comment:
:bulb: Add a test where this is used only in the constructor (allowed).

:bulb: Add matching tests for passing an `out` variable.